conversion.c

```diff
@@ -0,0 +1,122 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// Ayisha Sameera,GET.
+// ------------------------------------------------------------------
+// conversion.c
+// Functions to convert a decimal number to binary and hexadecimal.
+// ------------------------------------------------------------------------------------------------
+#pragma warning (disable:4996)
+#include <stdbool.h>
+#include <malloc.h>
+#include <limits.h>
+#include "conversion.h"
+#include <math.h>
+#include <string.h>
+#include <stdlib.h>
+
+// Returns the nearest bit length.
+int NearestBitLen (int digit) {
+   if (digit <= 8) return 8;               // 1 byte
+   else if (digit <= 16) return 16;        // 2 byte
+   else return 32;                         // 4 bytes, maximum value 
+}
+
+// Converts the given input to binary.
+char* Binary (char* input) {
+   long int num = 0, numCpy = num, power = 1;
+   int digit = 1, len, inInd = 0, lenIn = strlen (input);

```

Comment:
Better to name variables with names that convey their purpose. Idx, or Index is sufficient in this case.